app/v2/controllers/ErrorTranslator.scala

```diff
@@ -74,6 +75,7 @@ trait ErrorTranslator {
 
     override def convert(routerError: RouterError): PresentationError = routerError match {
       case err: RouterError.UnexpectedError => PresentationError.internalServiceError(cause = err.thr)
+      case UnrecognisedOffice(message)      => PresentationError.badRequestError(message)
```

Comment:
done the changes. will speak to Fergal if we need to change something else

Reply:
I discussed this with Fergal and he has suggested more generic way to do this. see below:

The customs office specified for `CustomsOfficeOfDestinationActual` or `CustomsOfficeOfDeparture` must be a customs office located in the United Kingdom.

His intention was to not use brackets and slash.

Please suggest if you are happy with above message? so that I can change it as per above suggestion given by Fergal

Reply:
Well, not really. I think the context of what I was saying was a bit lost here and .

The problem is that either `CustomsOfficeOfDeparture \ referenceNumber` or `CustomsOfficeOfDestinationActual \ referenceNumber` (the XML path to the element) must be in the UK, but it's not quite an either or, it's totally dependent on whether this is a departure (`CustomsOfficeOfDeparture`) or an arrival (`CustomsOfficeOfDestinationActual`). 

I'd written it the way I did as I was (badly!) trying to infer it's one of the other depending on this. We'll possibly need to have two error converters with two different messages for this reason.

Need to think on this tomorrow.

Reply:
I think we can display same messages whatever we receive from transit-movements-router in case of bad request to common-transit-convention-traders as well.
in our case with pending pull request of router I can see : **Did not recognise office: ${office.value}**
Better we can collect above same message and display with bad request response ? It's having users entered bad value so user should be able to identify what went wrong and then user can correct that?

Reply:
The problem with the router's message is that it doesn't specify which office went wrong, and this is going to be important particularly for arrivals, where they might have specified the office of destination and the office of destination actual as the same. The only one we care about for arrivals is the office of destination actual. Really, we need to specify the right one at the right time.

It's generally _not_ a good idea to pass through error messages from internal components (router) to API consumers because we do not want to accidently expose some internal messages. `common-transit-convention-traders` job is to turn internal error messages into ones that are appropriate for the user. If the router PR needs changes to support what you want to do, such as adding fields to the bad request response at the router level, feel free to ask for that! However, I don't want to pass through error messages unchecked, like you're proposing here.

For what it's worth, I've opened CTCP-1467 to put this check in a better place, but agreeing the wording now still makes sense.

Reply:
I think we can make decision based on the movementType (arrivals or departures) as we are passing movementType at the router level? 
anyway, do we need to put this on hold until decision is made using CTCP-1467?

Reply:
It depends on how quickly 1467 is going to come in and how we do it. I would have liked to discuss it in clarification today, alas, I have another meeting then.

Assume that 1467 isn't being done for now. As I said, the wording is important. And yes, we can use the movement type, but the message needs to be generated here.

Reply:
As discussed with Rob as of now I have made it generic message as suggested by Fergal.
I have created one ticket to track this issue CTCP-1505. That is dependent of CTCP-1467. Once CTCP-1467 is done. we will further decide what needs to be done as part of  CTCP-1505